Frameworks/Foundation/NSDecimal.mm

```diff
@@ -0,0 +1,589 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Starboard.h>
+#import <StubReturn.h>
+#import <Foundation/NSDecimal.h>
+
+#define IS_NEGATIVE(decimal) (decimal)->_isNegative
+
+#define IS_POSITIVE(decimal) !IS_NEGATIVE(decimal)
+
+#define IS_ZERO(decimal) (((decimal)->_length == 0) && IS_POSITIVE(decimal))
+
+#define INVALIDATE_DECIMAL(decimal) \
+    \
```

Comment:
So if you just delete these otherwise blank lines, clang-format adds them back?

---
In reply to: [80537387](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r80537387) [](ancestors = 80537387,80536986,80341299,80341159)